src/packet/pacer.rs

```diff
@@ -464,62 +450,39 @@ impl LeakyBucketPacer {
-                        .unwrap_or(now),
-                    Some(queue),
-                    None,
-                )
-            }
-            None if self.padding_debt == DataSize::ZERO
-                && self.media_debt == DataSize::ZERO
-                && self.padding_bitrate > Bitrate::ZERO
-                && self.last_send_time.is_some() =>
-            {
-                // If we can generate padding do so on the most recently used queue.
-                let queue = self
-                    .queue_states
-                    .iter()
-                    .filter(|q| q.use_for_padding)
-                    .max_by_key(|q| q.snapshot.last_emitted);
-
-                // No queues and no debt, generate some padding.
-                let padding_to_add = self.padding_bitrate * PADDING_BURST_INTERVAL;
-
-                (now, queue, Some(padding_to_add))
+                let poll_at = self
+                    .last_handle_time
+                    .map(|h| h + next_send_offset)
+                    .unwrap_or(now);
+
+                return Some((poll_at, queue));
             }
-            None if self.padding_bitrate > Bitrate::ZERO && self.last_send_time.is_some() => {
+        }
+
+        if self.padding_bitrate > Bitrate::ZERO && self.last_send_time.is_some() {
+            let padding_queue = self
+                .queue_states
+                .iter()
+                .filter(|q| q.use_for_padding)
+                .max_by_key(|q| q.snapshot.last_emitted);
+            if let Some(queue) = padding_queue {
                 // If all queues are empty, we have sent something, and we have a padding rate wait until we have drained
                 // both the media debt and padding debt to send some padding.
                 let mut drain_debt_time = (self.media_debt / self.adjusted_bitrate)
                     .max(self.padding_debt / self.padding_bitrate);
-                let padding_queue = self
-                    .queue_states
-                    .iter()
-                    .filter(|q| q.use_for_padding)
-                    .max_by_key(|q| q.snapshot.last_emitted);
-
-                (
-                    self.last_handle_time
-                        .map(|h| h + drain_debt_time)
-                        .unwrap_or(now),
-                    padding_queue,
-                    // we return None here because we'll eventually get a handle_timeout that
-                    // will clear out the debt and we can nominate a padding size
-                    None,
-                )
-            }
-            _ => {
-                // No queues to send on, no padding to generate, wait until the next handle time or
-                // until a packet is queued.
-                (
-                    self.last_handle_time.map(|h| h + PACING).unwrap_or(now),
-                    None,
-                    None,
-                )
+                assert!(drain_debt_time > Duration::ZERO);
```

Comment:
Should get rid of this assert, but if this does happen we should return a timeout that's like `1us` so the main loop can have time to queue media if required